### ISSUE TITLE
Bad permission for salt group

### ISSUE THREAD

**Issue by speedytom** [saltstack/salt#65164]
**Description**
I have problem with our GitHub runner. It runs commands by ssh:
root@salt:/var/log/salt# sudo codac -c "salt -L 'dtacqpc,epicsiocs,minispecs,epicstest' -a pam --username=codac --password=XXXXXXXXXX state.apply codac.deployment"

root@salt:~# id codac
uid=599(codac) gid=100(users) groups=100(users),128(salt)

It doesn't work after upgrade to the newer version. I fixed this way:
<pre>
root@salt:/srv# chown salt:salt -R /var/run/salt/
root@salt:/srv# chown salt:salt -R /var/cache/salt
root@salt:/srv# chown -R salt:salt /var/log/salt/
</pre>

**Setup**
<pre>
root@salt:~# dpkg -l "salt*"
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version      Architecture Description
+++-==============-============-============-================================================================
ii  salt-api       3006.3       amd64        Generic, modular network access system
ii  salt-common    3006.3       amd64        shared libraries that salt requires for all packages
ii  salt-master    3006.3       amd64        remote manager to administer servers via salt
ii  salt-minion    3006.3       amd64        client package for salt, the distributed remote execution system
ii  salt-ssh       3006.3       amd64        remote manager to administer servers via Salt SSH
</pre>

**Steps to Reproduce the behavior**
I got this error before my fix:
root@salt:/var/log/salt# sudo codac -c "salt -L 'dtacqpc,epicsiocs,minispecs,epicstest' -a pam --username=codac --password=XXXXXXXXXX state.apply codac.deployment"
<pre>
write(1, "\33[0;32mSalt request timed out. T"..., 372Salt request timed out. The master is not responding. You may need to run your command with `--async` in order to bypass the congested event bus. With `--async`, the CLI tool will print the job id (jid) and exit immediately without listening for responses. You can then use `salt-run jobs.lookup_jid` to look up the results of the job in the job cache later.
) = 372
</pre>

**Expected behavior**

orig:
postinst:
    chown -R salt:salt /etc/salt/pki/master /etc/salt/master.d /var/log/salt/master /var/log/salt/key /var/cache/salt/master /var/run/salt/master

fixed:
postinst:
chown -R salt:salt /etc/salt/pki/master /etc/salt/master.d /var/log/salt /var/log/salt/key /var/cache/salt /var/run/salt

**Comment by anilsil** [comment]
@speedytom Can you please update the version of salt ?

**Comment by speedytom** [comment]
You can see in the setup.

I have to change salt path manually by the way:
<pre>
root@salt:~# salt --versions-report
Salt Version:
          Salt: 3006.2
 
Python Version:
        Python: 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0]
 
Dependency Versions:
          cffi: Not Installed
      cherrypy: Not Installed
      dateutil: 2.8.1
     docker-py: Not Installed
         gitdb: 4.0.9
     gitpython: 3.1.24
        Jinja2: 3.0.3
       libgit2: 1.1.0
  looseversion: 1.3.0
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.5
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     packaging: 21.3
     pycparser: Not Installed
      pycrypto: Not Installed
  pycryptodome: 3.18.0
        pygit2: 1.6.1
  python-gnupg: Not Installed
        PyYAML: 5.4.1
         PyZMQ: 25.0.2
        relenv: Not Installed
         smmap: 5.0.0
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.4
 
System Versions:
          dist: ubuntu 22.04.3 jammy
        locale: utf-8
       machine: x86_64
       release: 5.15.0-79-generic
        system: Linux
       version: Ubuntu 22.04.3 jammy

root@salt:~# /usr/bin/salt --versions-report
Salt Version:
          Salt: 3006.3
 
Python Version:
        Python: 3.10.13 (main, Sep  6 2023, 02:11:27) [GCC 11.2.0]
 
Dependency Versions:
          cffi: 1.14.6
      cherrypy: unknown
      dateutil: 2.8.1
     docker-py: Not Installed
         gitdb: 4.0.10
     gitpython: 3.1.31
        Jinja2: 3.1.2
       libgit2: 1.1.0
  looseversion: 1.0.2
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     packaging: 22.0
     pycparser: 2.21
      pycrypto: Not Installed
  pycryptodome: 3.9.8
        pygit2: 1.6.1
  python-gnupg: 0.4.8
        PyYAML: 6.0.1
         PyZMQ: 23.2.0
        relenv: 0.13.10
         smmap: 5.0.0
       timelib: 0.2.4
       Tornado: 4.5.3
           ZMQ: 4.3.4
 
System Versions:
          dist: ubuntu 22.04.3 jammy
        locale: utf-8
       machine: x86_64
       release: 5.15.0-79-generic
        system: Linux
       version: Ubuntu 22.04.3 jammy
 
root@salt:~# diff -u /usr/bin/salt /usr/local/bin/salt
--- /usr/bin/salt	2023-09-06 18:51:25.000000000 +0200
+++ /usr/local/bin/salt	2023-09-04 09:09:45.371323892 +0200
@@ -1,7 +1,4 @@
-#!/bin/sh
-"true" ''''
-"exec" "$(dirname "$(readlink -f "$0")")/bin/python3.10" "$0" "$@"
-'''
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 import re
 import sys

root@salt:~# mv /usr/local/bin/salt /usr/local/bin/salt.bad
root@salt:/usr/local/bin# ln -s /usr/bin/salt salt

</pre>

**Comment by Ch3LL** [comment]
This PR https://github.com/saltstack/salt/pull/64194 made some changes in 3006.3

The key changes here are the following: 

```
Under /var/cache/salt and /var/run/salt only give ownership on master directories
Under /var/log/salt, ensure /var/log/salt/master exists and give ownership of that. Also update logrotate config to create that with correct ownership and perms and install that on debian packages.
```

What permissions did you have for the master directory under each of those directories?

**Comment by speedytom** [comment]
I suppose I have root:root for these directories.

**Comment by speedytom** [comment]
I change only owner and group for t
root@salt:~# stat /var/cache/salt
  File: /var/cache/salt
  Size: 4096      	Blocks: 8          IO Block: 4096   directory
Device: fc02h/64514d	Inode: 2884020     Links: 5
Access: (0775/drwxrwxr-x)  Uid: (  996/    salt)   Gid: (  128/    salt)
Access: 2023-09-14 08:03:29.992503419 +0200
Modify: 2023-06-06 14:49:33.588049009 +0200
Change: 2023-09-11 11:58:24.178168599 +0200
 Birth: 2021-03-15 07:26:36.632656547 +0100
root@salt:~# 
root@salt:~# 
root@salt:~# stat /var/run/salt
  File: /var/run/salt
  Size: 60        	Blocks: 0          IO Block: 4096   directory
Device: 18h/24d	Inode: 1289        Links: 3
Access: (0755/drwxr-xr-x)  Uid: (  996/    salt)   Gid: (  128/    salt)
Access: 2023-09-11 13:15:23.414952070 +0200
Modify: 2023-08-16 09:23:39.967672724 +0200
Change: 2023-09-11 11:58:19.282155048 +0200

**Comment by speedytom** [comment]
Salt-master change ownership again itself:
<pre>
root@salt:~# ls -la /var/run/salt
total 0
drwxr-xr-x  3 salt salt  60 Aug 16 09:23 .
drwxr-xr-x 30 root root 980 Sep 14 08:08 ..
drwxr-xr-x  2 root root 120 Sep 12 09:00 master
root@salt:~# ls -la /var/cache/salt
total 20
drwxrwxr-x  5 salt salt 4096 Jun  6 14:49 .
drwxr-xr-x 15 root root 4096 Mar 15  2021 ..
drwxrwxr-x 16 root root 4096 Sep 12 09:00 master
drwxrwxr-x  6 salt salt 4096 Oct  7  2022 minion
drwxrwxr-x  6 salt salt 4096 Jun  6 15:10 proxy
</pre>

????

**Comment by speedytom** [comment]
I suppose that salt-master should work as other Linux services under specific user. In this case salt.
<pre>
root@salt:~# ps -ef |grep salt
root     2833706       1  0 Sep08 ?        00:00:09 /opt/saltstack/salt/bin/python3.10 /usr/bin/salt-api MainProcess
root     2833777 2833706  0 Sep08 ?        01:00:02 /opt/saltstack/salt/bin/python3.10 /usr/bin/salt-api RunNetapi(salt.loaded.int.netapi.rest_cherrypy)
root     3466332       1  0 08:28 ?        00:00:00 /opt/saltstack/salt/bin/python3.10 /usr/bin/salt-master MainProcess
root     3466381 3466332  0 08:28 ?        00:00:00 /opt/saltstack/salt/bin/python3.10 /usr/bin/salt-master PubServerChannel._publish_daemon
root     3466382 3466332  0 08:28 ?        00:00:00 /opt/saltstack/salt/bin/python3.10 /usr/bin/salt-master EventPublisher
</pre>

**Comment by speedytom** [comment]
<pre>
root@salt:~# cat /lib/systemd/system/salt-master.service
[Unit]
Description=The Salt Master Server
Documentation=man:salt-master(1) file:///usr/share/doc/salt/html/contents.html https://docs.saltproject.io/en/latest/contents.html
After=network.target

[Service]
LimitNOFILE=100000
Type=notify
NotifyAccess=all
ExecStart=/usr/bin/salt-master

[Install]
WantedBy=multi-user.target
</pre>

**Comment by speedytom** [comment]
I changed salt-master to salt:
<pre>
root@salt:/srv/salt# cat /etc/salt/master |grep user
# The user under which the salt master will run. Salt will update all
# permissions to allow the specified user to run the master. The exception is
# the job cache, which must be deleted if this user is changed. If the
user: salt
</pre>

I removed user codac and using user salt for CI GigLab deployment.  
ssh -T codac@salt.tok.ipp.cas.cz "${NODES}

 sudo codac -c "salt -L 'dtacqpc,epicsiocs,minispecs,epicstest' -a pam --username=codac --password=XXXXXXXXXX state.apply codac.deployment"
to 
ssh -T salt@salt.tok.ipp.cas.cz "${NODES}
I don't need sudo anymore.

I fix my problem by the way. 

I suppose there is one error after upgrade as I mentioned
<pre>
root@salt:~# diff -u /usr/bin/salt /usr/local/bin/salt
--- /usr/bin/salt	2023-09-06 18:51:25.000000000 +0200
+++ /usr/local/bin/salt	2023-09-04 09:09:45.371323892 +0200
@@ -1,7 +1,4 @@
-#!/bin/sh
-"true" ''''
-"exec" "$(dirname "$(readlink -f "$0")")/bin/python3.10" "$0" "$@"
-'''
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 import re
 import sys

root@salt:~# mv /usr/local/bin/salt /usr/local/bin/salt.bad
root@salt:/usr/local/bin# ln -s /usr/bin/salt salt
</pre>

**Comment by dmurphy18** [comment]
@speedytom This issue is being addressed by PR https://github.com/saltstack/salt/pull/66218 which is nearing completion

**Comment by dmurphy18** [comment]
@speedytom Closing this issue with https://github.com/saltstack/salt/pull/66218 now merged into 3006.x branch

**Comment by jamest-pin** [comment]
@dmurphy I'm having issues after upgrading to salt 3007 from 3005.
We have CI runner user (in our case `buildkite-agent` which is configured to be allowed to run salt calls.
Prior to update, this was working.
Now after updating, we get this error 
```
$ salt 'salt' test.ping
[WARNING ] Failed to open log file, do you have permission to write to /var/log/salt/master?
[WARNING ] TCP Publish Client encountered an exception while connecting to /var/run/salt/master/master_event_pub.ipc: StreamClosedError('Stream is closed'), will reconnect in 1 seconds -   File "/usr/bin/salt", line 11, in <module>
    sys.exit(salt_main())

  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/scripts.py", line 528, in salt_main
    client.run()

  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/cli/salt.py", line 192, in run
    for full_ret in cmd_func(**kwargs):

  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/client/__init__.py", line 815, in cmd_cli
    self.pub_data = self.run_job(

  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/client/__init__.py", line 387, in run_job
    pub_data = self.pub(

  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/client/__init__.py", line 1904, in pub
    if listen and not self.event.connect_pub(timeout=timeout):

  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/utils/event.py", line 323, in connect_pub
    self.subscriber = salt.utils.asynchronous.SyncWrapper(

  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/utils/asynchronous.py", line 76, in __init__
    self.obj = cls(*args, **kwargs)

  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/transport/base.py", line 210, in ipc_publish_client
    return publish_client(opts, io_loop, **kwargs)

  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/transport/base.py", line 152, in publish_client
    return salt.transport.tcp.PublishClient(

  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/transport/tcp.py", line 220, in __init__
    super().__init__(opts, io_loop, **kwargs)

  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/transport/base.py", line 398, in __init__
    super().__init__()

Authentication error occurred.
```
permissions:
```
james_tuson@salt:~$ ll /var/log/salt/master
-rw-r----- 1 salt salt 990961 Jul 24 07:55 /var/log/salt/master
james_tuson@salt:~$ ll /var/run/salt/master
total 0
drwxr-xr-x 2 salt salt 120 Jul 24 07:28 ./
drwxr-xr-x 4 root root  80 Jul 24 04:32 ../
srw------- 1 salt salt   0 Jul 24 07:28 master_event_pub.ipc=
srw------- 1 salt salt   0 Jul 24 07:28 master_event_pull.ipc=
srw------- 1 salt salt   0 Jul 24 07:28 publish_pull.ipc=
srw------- 1 salt salt   0 Jul 24 07:28 workers.ipc=
```

config:
```yaml
salt@salt:/etc/salt$ cat master | grep buildkite -A 3 -B 3
#    - network.*
#
client_acl:
  buildkite-agent:
    - .*

publisher_acl:
  buildkite-agent:
    - .*

external_auth:
  pam:
    buildkite-agent:
      - .*

```

**Comment by jamest-pin** [comment]
Updating permissions as per https://docs.saltproject.io/en/latest/ref/publisheracl.html#publisher-acl-system fixed this `unable to access /varlog/salt/master` but I shouldn't have to fix perms after an upgrade.

**Comment by dmurphy18** [comment]
@jamest-pin The fix #66218 will be available in 3006.9 (releasing very soon) it is not yet available in 3007.x.

**Comment by jamest-pin** [comment]
Noted, thanks.
